numpyro/infer/mcmc.py

```diff
@@ -338,7 +338,10 @@ def _single_chain_mcmc(self, init, args, kwargs, collect_fields):
         states = dict(zip(collect_fields, states))
         # Apply constraints if number of samples is non-zero
         site_values = tree_flatten(states[self._sample_field])[0]
-        if len(site_values) > 0 and site_values[0].size > 0:
+        # XXX: lax.map still works if some arrays have 0 size
+        # so we only need to filter out the case site_value.shape[0] == 0
+        # (which happens when lower_idx==upper_idx)
+        if len(site_values) > 0 and jnp.shape(site_values[0])[0] > 0:
```

Comment:
@neerajprad Could you help me review this change? If we use `site_values[0].size` here, those values with shape `(num_samples, 0)` will not be post-processed, hence some errors will happen when we use the samples for other tasks (or even when we print the summary). For simplex constraint (the support of Dirichlet), if the constrained value has shape (1,), then its unconstrained value will have shape (0,). That will give use constant constrained samples. I believe this is not useful, but it is better to let users decide if they want to do so, rather than just throwing a misleading error e.g. `ValueError: cannot reshape array of size 0 into shape (1,0)` when I summarize a `hmm` model with this prior.